helix-core/src/increment/integer.rs

```diff
@@ -10,37 +10,45 @@ use crate::{
 };
 
 #[derive(Debug, PartialEq, Eq)]
-pub struct NumberIncrementor<'a> {
+pub struct IntegerIncrementor<'a> {
     value: i64,
     radix: u32,
     range: Range,
 
     text: RopeSlice<'a>,
 }
 
-impl<'a> NumberIncrementor<'a> {
-    /// Return information about number under rang if there is one.
-    pub fn from_range(text: RopeSlice, range: Range) -> Option<NumberIncrementor> {
-        // If the cursor is on the minus sign of a number we want to get the word textobject to the
-        // right of it.
+impl<'a> IntegerIncrementor<'a> {
+    /// Return information about an integer under range if there is one.
+    /// Integer includes possible negative sign, digits in bases 2, 8, 10, or 16 and underscores.
+    /// Integer does not include decimal point or trailing underscores
+    /// Examples:
+    ///    -1, 99, 0xABCD, 0x1010, 0o1234567, 1_000_000
+    pub fn from_range(text: RopeSlice, range: Range) -> Option<IntegerIncrementor> {
+        // If the cursor is on the minus sign of an integer we want to get the word textobject
+        // to the right of it.
         let range = if range.to() < text.len_chars()
-            && range.to() - range.from() <= 1
+            && range.len() <= 1
             && text.char(range.from()) == '-'
         {
             Range::new(range.from() + 1, range.to() + 1)
         } else {
             range
         };
 
-        let range = textobject_word(text, range, TextObject::Inside, 1, false);
+        let mut range = textobject_word(text, range, TextObject::Inside, 1, false);
 
         // If there is a minus sign to the left of the word object, we want to include it in the range.
-        let range = if range.from() > 0 && text.char(range.from() - 1) == '-' {
-            range.extend(range.from() - 1, range.from())
-        } else {
-            range
+        if range.from() > 0 && text.char(range.from() - 1) == '-' {
+            range = range.extend(range.from() - 1, range.from())
         };
 
+        // If there are trailing underscores, remove them from the range.
+        // There is a `- 1` here because range ends are exclusive.
+        while !range.is_empty() && text.char(range.to() - 1) == '_' {
+            range = range.shorten()
```

Comment:
`Range::shorten` is a bit confusing: it's not clear which end is being trimmed and by how much, and it can produce a range with 0 width. I think we can get away with just tracking `to` in this loop and not creating a new range for each iteration, and returning `None` if after this loop `to <= range.from()`.